vars/getFunctionalTestStage.groovy

```diff
@@ -0,0 +1,91 @@
+ *      name            functional test stage name
+ *      pragma_suffix   functional test stage commit pragma suffix 
+ *      label           functional test stage default cluster label
+ *      next_version    next daos package version
+ *      stage_tags      functional test stage tags always used and combined with all other tags
+ *      timer_tags      functional test stage tags to use when the stage is started by a timer
+ *      default_tags    launch.py tags argument to use when no parameter or commit pragma exist
+ *      default_nvme    launch.py --nvme argument to use when no parameter or commit pragma exist
+ *      provider        launch.py --provider argument to use
+ *      distro          functional test stage distro (VM)
+ *      job_status      Map of status for each stage in the job/build
+ * @return a scripted stage to run in a pipeline
+ */
+Map call(Map kwargs = [:]) {
+    String name = kwargs.get('name')
+    String pragma_suffix = kwargs.get('pragma_suffix')
+    String label = kwargs.get('label')
+    String next_version = kwargs.get('next_version', null)
+    String stage_tags = kwargs.get('stage_tags')
+    String timer_tags = kwargs.get('timer_tags')
+    String default_tags = kwargs.get('default_tags')
+    String default_nvme = kwargs.get('default_nvme')
+    String provider = kwargs.get('provider', '')
+    String distro = kwargs.get('distro')
+    Map job_status = kwargs.get('job_status', [:])
+
+    return {
+        stage("${name}") {
+            // Get the tags for thge stage. Use the timer_tags if the build has been started by a
+            // timer. Otherwise use either the build parameter, commit pragma, or default tags. All
+            // tags are comnbined with the stage tags to ensure only tests that 'fit' the cluster
+            // will be run.
+            if (startedByTimer() && timer_tags) {
+                default_tags = timer_tags
+            }
+            String tags = getFunctionalTags(
+                pragma_suffix: pragma_suffix, stage_tags: stage_tags, default_tags: default_tags)
+
+            // Setup the arguments for the skipStage() groovy script to directly call the correct
+            // skip stage logic. The stage name is no longer required to be defined in skipStage().
+            Map skip_config = ['tags': tags]
+            if (pragma_suffix.startsWith('hw-')) {
+                // With this param set skip_ftest_hw() will be called by skipStage()
+                skip_config['hw_size'] = pragma_suffix.replace('hw-', '')
+            } else if (kwargs['distro']) {
+                // With this param set skip_ftest() will be called by skipStage()
+                skip_config['distro'] = kwargs['distro']
+            }
+
+            echo "[${name}] Start with ${skip_config}"
+            if (skipStage(skip_config)) {
+                echo "[${name}] Stage skipped by skipStage(${skip_config})"
```

Comment:
Hrm.  I'm not sure I am a fan of skipped stages showing green rather than the "empty" that they currently display as.

Reply:
Yeah, me either, but I think I found a way to do it properly.